### PR TITLE
Updated POJOs, renamed intro_screen_two to activity_sign_in

### PR DESCRIPTION
**Pull request by drulabs**
Fixes # NA

## Description

Refactored starter code. Renamed layouts as per our naming conventions, made changes in POJO as per discussion. like adding bookmarks field in User POJO.

## Before & After Screenshots

**BEFORE**:
NA

**AFTER**:
NA

## Other changes (e.g. bug fixes, UI tweaks, refactors)

None

### CODE REVIEW

**Comment by drulabs** [comment]
Please merge this ASAP. A little bit of refactoring of starter code.

File: app/src/main/res/layout/activity_sign_in.xml [hunk 1]
```diff
@@ -1,35 +1,50 @@
...
 
-    <EditText
-        android:id="@+id/editText"
+
+    <ImageView
+        android:id="@+id/sigin_image_logo"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:ems="10"
-        android:hint="@string/hint_email"
```
Comment (sagarsrao) [inline app/src/main/res/layout/activity_sign_in.xml]: Kindly add tools:text for the characters visibility
Go through the below link
https://developer.android.com/studio/write/tool-attributes.html

File: app/src/main/res/layout/activity_sign_in.xml [hunk 2]
```diff
@@ -1,35 +1,50 @@
...
-
-    <EditText
-        android:id="@+id/editText2"
+        android:layout_marginTop="80dp"
+        android:layout_centerHorizontal="true"
+        android:src="@mipmap/signin_logo" />
+
+    <TextView
+        android:id="@+id/signin_text_almost_there"
+        android:layout_marginTop="16dp"
```
Comment (sagarsrao) [inline app/src/main/res/layout/activity_sign_in.xml]: please remove hardcoded string and do take it from dimes.xml file

File: app/src/main/res/layout/activity_sign_in.xml [hunk 3]
```diff
@@ -1,35 +1,50 @@
...
+        android:fontFamily="@font/roboto_medium"
+        android:text="@string/sigin_text_almost_there"
+        android:textColor="#90ffffff" />
+
+    <com.google.android.gms.common.SignInButton
+        android:id="@+id/google_signin_btn"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/signin_text_almost_there"
+        android:layout_marginEnd="32sp"
```
Comment (sagarsrao) [inline app/src/main/res/layout/activity_sign_in.xml]: No Hardcoded strings in the xml file should be there please take it from dimens.xml file

File: app/src/main/res/layout/activity_sign_in.xml [hunk 4]
```diff
@@ -1,35 +1,50 @@
...
+        android:layout_marginTop="100dp">
+
+    </com.google.android.gms.common.SignInButton>
+
+    <ProgressBar
+        android:id="@+id/progressbar_google_signin"
+        style="?android:attr/progressBarStyle"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginBottom="8dp"
```
Comment (sagarsrao) [inline app/src/main/res/layout/activity_sign_in.xml]: Kindly make a common size constraints in the dimens.xml file and we can use it everywhere

File: app/src/main/res/layout/intro_screen_two.xml [hunk 5]
```diff
@@ -1,50 +0,0 @@
...
-    <TextView
-        android:id="@+id/signin_text_almost_there"
-        android:layout_marginTop="16dp"
-        android:layout_below="@id/sigin_image_logo"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_centerHorizontal="true"
-        android:fontFamily="@font/roboto_medium"
-        android:text="@string/sigin_text_almost_there"
-        android:textColor="#90ffffff" />
```
Comment (sagarsrao) [inline app/src/main/res/layout/intro_screen_two.xml]: text color should be taken from colors.xml file

File: app/src/main/res/layout/intro_screen_two.xml [hunk 6]
```diff
@@ -1,50 +0,0 @@
...
-
-    <com.google.android.gms.common.SignInButton
-        android:id="@+id/google_signin_btn"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_marginEnd="32sp"
```
Comment (sagarsrao) [inline app/src/main/res/layout/intro_screen_two.xml]: Please remove hardcoded strings from the file and take it from dimens.xml file

Comment (drulabs) [reply app/src/main/res/layout/intro_screen_two.xml]: red colour means deleted. you are commenting on a deleted file.

File: app/src/main/res/layout/intro_screen_two.xml (hunk 5 above)
Comment (drulabs) [reply app/src/main/res/layout/intro_screen_two.xml]: red colour means deleted. you are commenting on a deleted file.

File: app/src/main/res/layout/activity_sign_in.xml (hunk 4 above)
Comment (drulabs) [reply app/src/main/res/layout/activity_sign_in.xml]: red colour means deleted. you are commenting on a deleted file.

File: app/src/main/java/com/developervishalsehgal/udacityscholarsapp/data/models/Discussion.java [hunk 7]
```diff
@@ -10,9 +10,6 @@
  */
 public class Discussion {
 
-    @SerializedName("deadline")
-    private String mDeadline;
```
[the remark this answers is not in the document]
Comment (drulabs) [reply app/src/main/java/com/developervishalsehgal/udacityscholarsapp/data/models/Discussion.java]: file is deleted

File: app/src/main/res/layout/activity_sign_in.xml (hunk 1 above)
Comment (drulabs) [reply app/src/main/res/layout/activity_sign_in.xml]: file is deleted

File: app/src/main/res/layout/activity_sign_in.xml (hunk 2 above)
Comment (drulabs) [reply app/src/main/res/layout/activity_sign_in.xml]: These changes are from #10 from @derrickabreo, he made the commits before we updated our coding style guidelines. This is not a shows stopper or breaking change. We can take it up later, right now the priority is  making the development branch stable.  @sagarsrao please create an issue and note these points there.

File: app/src/main/res/layout/activity_sign_in.xml (hunk 3 above)
Comment (drulabs) [reply app/src/main/res/layout/activity_sign_in.xml]: These changes are from #10 from @derrickabreo, he made the commits before we updated our coding style guidelines. This is not a shows stopper or breaking change. We can take it up later, right now the priority is  making the development branch stable.  @sagarsrao please create an issue and note these points there.

File: app/src/main/java/com/developervishalsehgal/udacityscholarsapp/data/models/Discussion.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (sagarsrao) [reply app/src/main/java/com/developervishalsehgal/udacityscholarsapp/data/models/Discussion.java]: Sorry for that

File: app/src/main/res/layout/activity_sign_in.xml (hunk 1 above)
Comment (sagarsrao) [reply app/src/main/res/layout/activity_sign_in.xml]: We can think of using in existing xml files

File: app/src/main/res/layout/activity_sign_in.xml (hunk 2 above)
Comment (sagarsrao) [reply app/src/main/res/layout/activity_sign_in.xml]: Will do that.Thanks

File: app/src/main/res/layout/intro_screen_two.xml (hunk 6 above)
Comment (sagarsrao) [reply app/src/main/res/layout/intro_screen_two.xml]: Sorry for that

**Pull request merged**